nullptr after coming back after 1 hour

Exception thrown by application class 'javax.faces.webapp.FacesServlet.service:230'
javax.servlet.ServletException: javax.el.ELException: java.lang.NullPointerException
at javax.faces.webapp.FacesServlet.service(FacesServlet.java:230)
at com.ibm.ws.webcontainer.servlet.ServletWrapper.service(ServletWrapper.java:1287)
at [internal classes]
at dv512.controller.filter.AuthenticationFilter.doFilter(AuthenticationFilter.java:45)
at com.ibm.ws.webcontainer.filter.FilterInstanceWrapper.doFilter(FilterInstanceWrapper.java:207)
at [internal classes]
at dv512.controller.filter.IndexFilter.doFilter(IndexFilter.java:49)
at com.ibm.ws.webcontainer.filter.FilterInstanceWrapper.doFilter(FilterInstanceWrapper.java:207)
at [internal classes]
Caused by: org.apache.myfaces.view.facelets.el.ContextAwareELException: javax.el.ELException: java.lang.NullPointerException
at org.apache.myfaces.view.facelets.el.ContextAwareTagMethodExpression.invoke(ContextAwareTagMethodExpression.java:108)
... 7 more
Caused by: javax.el.ELException: java.lang.NullPointerException
at org.apache.el.parser.AstValue.invoke(AstValue.java:258)
... 7 more
Caused by: java.lang.NullPointerException: 
at dv512.controller.FeedController.loadData(FeedController.java:80)
at sun.reflect.GeneratedMethodAccessor115.invoke(Unknown Source)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
at java.lang.reflect.Method.invoke(Unknown Source)
at org.apache.el.parser.AstValue.invoke(AstValue.java:245)
... 7 more

--- a/implementation/2dv512-app/WebContent/template/sections/header.xhtml
+++ b/implementation/2dv512-app/WebContent/template/sections/header.xhtml
@@ -12,8 +12,8 @@
 			<div id="content">	
 				<h:graphicImage id="icon" library="img" name="ic_dog.png" />
 				<h1><a href="#{request.contextPath}/index.xhtml" style="text-decoration: none; color: inherit;">Internet of Dogs</a></h1>
-				<h:form id="logout" rendered="#{loginController.verified}">
-					<h:commandLink value="Sign Out" action="#{loginController.logout}" /><i class="material-icons md-18">exit_to_app</i>
+				<h:form id="logout" rendered="#{userSession.valid}">
+					<h:commandLink value="Sign Out" action="#{userSession.logout}" /><i class="material-icons md-18">exit_to_app</i>
 				</h:form>
 			</div>
 		</header>

--- a/implementation/2dv512-app/src/dv512/UserSession.java
+++ b/implementation/2dv512-app/src/dv512/UserSession.java
@@ -7,6 +7,7 @@ import java.time.Instant;
 import java.util.Date;
 import java.util.TimeZone;
 
+import javax.annotation.PreDestroy;
 import javax.enterprise.context.SessionScoped;
 import javax.inject.Inject;
 import javax.inject.Named;
@@ -21,6 +22,8 @@ import dv512.model.dao.ProfilesDAO;
 public class UserSession implements Serializable {
 	private static final long serialVersionUID = 1L;
 
+	public static final String ACTION_LOGOUT = "logout";
+	
 	@Inject
 	private ProfilesDAO profiles;
 	
@@ -43,7 +46,7 @@ public class UserSession implements Serializable {
 		this.userId = id;
 		reload();
 	}
-		
+			
 	public void setTimeZone(TimeZone timeZone) {
 		this.timeZone = timeZone;
 		
@@ -71,6 +74,10 @@ public class UserSession implements Serializable {
 		return notificationCount;
 	}
 	
+	public boolean isValid() {
+		return userId != User.UNKNOWN_ID;
+	}
+	
 	
 	public void reload() {
 		if(userId != -1) {
@@ -81,5 +88,20 @@ public class UserSession implements Serializable {
 		else {
 			profile = null;
 		}
+	}
+		
+	public String logout() {		
+		userId = User.UNKNOWN_ID;
+		profile = null;
+		timeZone = null;
+		lastNotificationCountUpdateTime = 0;
+		notificationCount = 0;
+		return ACTION_LOGOUT;
+	}
+	
+	@PreDestroy
+	private void onDestroy() {
+		System.out.println("User session detroyed.");
+		logout();		
 	}
 }

--- a/implementation/2dv512-app/src/dv512/controller/LoginController.java
+++ b/implementation/2dv512-app/src/dv512/controller/LoginController.java
@@ -6,8 +6,8 @@ import java.io.Serializable;
 import java.util.Properties;
 import java.util.TimeZone;
 
-import javax.enterprise.context.ApplicationScoped;
 import javax.faces.context.FacesContext;
+import javax.faces.view.ViewScoped;
 import javax.inject.Inject;
 import javax.inject.Named;
 import javax.mail.Authenticator;
@@ -28,7 +28,7 @@ import dv512.model.dao.UsersDAO;
 
 
 @Named
-@ApplicationScoped
+@ViewScoped
 public class LoginController implements Serializable {
 	private static final long serialVersionUID = 1L;
 
@@ -38,20 +38,18 @@ public class LoginController implements Serializable {
 	
 	public static final String ACTION_LOGIN_SUCCESS = "success";
 	public static final String ACTION_LOGIN_FAIL = "fail";	
-	public static final String ACTION_LOGOUT = "logout";
+
 	
 	@Inject
 	private UserSession session;
 	
 	@Inject
 	private UsersDAO userDAO;
-	
-	
+		
 	private int currentState = STATE_LOGIN;
 	
 	private User user;
-	private int retryCount = 0;
-	
+	private int retryCount = 0;	
 	private int timeZoneOffset = 0;
 	
 	public LoginController() {
@@ -73,11 +71,7 @@ public class LoginController implements Serializable {
 	public int getRetryCount() {
 		return retryCount;
 	}
-	
-	public boolean isVerified() {
-	    return user.getId() != User.UNKNOWN_ID;
-	}
-	
+		
 	public int getUserId() {
 		return user.getId();
 	}
@@ -100,17 +94,7 @@ public class LoginController implements Serializable {
 			return ACTION_LOGIN_FAIL;
 		}
 	}
-	
-	public String logout() {
-		user.setId(User.UNKNOWN_ID);
-		session.setUserId(User.UNKNOWN_ID);
-		retryCount = 0;
-		user.setEmail(null);
-		user.setPassword(null);		
-		return ACTION_LOGOUT;
-	}
-	
-	
+
 	public String sendResetEmail() throws IOException, AddressException, MessagingException {		
 		String token = userDAO.requestResetPassword(user);
 		
@@ -176,6 +160,4 @@ public class LoginController implements Serializable {
 		currentState = STATE_LOGIN;
 	}
 
-	
-	
 }

--- a/implementation/2dv512-app/src/dv512/controller/filter/AuthenticationFilter.java
+++ b/implementation/2dv512-app/src/dv512/controller/filter/AuthenticationFilter.java
@@ -13,13 +13,13 @@ import javax.servlet.annotation.WebFilter;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-import dv512.controller.LoginController;
+import dv512.UserSession;
 
 @WebFilter("/user/*")
 public class AuthenticationFilter implements Filter {
 
 	@Inject
-	private LoginController loginController;
+	private UserSession session;
 	
 
 	@Override
@@ -37,7 +37,7 @@ public class AuthenticationFilter implements Filter {
 		String servPath = request.getServletPath();
 		System.out.println("AuthenticationFilter: " + servPath);
 		
-		if (!loginController.isVerified()) {
+		if (!session.isValid()) {
 			// redirect to login page, user not verified.
 			response.sendRedirect(request.getContextPath() + "/index.xhtml");
 		}

--- a/implementation/2dv512-app/src/dv512/controller/filter/IndexFilter.java
+++ b/implementation/2dv512-app/src/dv512/controller/filter/IndexFilter.java
@@ -13,13 +13,13 @@ import javax.servlet.annotation.WebFilter;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-import dv512.controller.LoginController;
+import dv512.UserSession;
 
 @WebFilter("/*")
 public class IndexFilter implements Filter {
 
 	@Inject
-	private LoginController loginController;
+	private UserSession session;
 	
 	
 	@Override
@@ -40,7 +40,7 @@ public class IndexFilter implements Filter {
 			return;
 		}
 		else if(servPath.equals("/index.xhtml")) {
-			if(loginController.isVerified()) {
+			if(session.isValid()) {
 				response.sendRedirect(request.getContextPath() + "/user/feed.xhtml");
 				return;
 			}

